### PR TITLE
Expanded

### PR DESCRIPTION
**Pull request by expede**
This is me just playing around with the existing code. Nothing too radical here, mostly letting the OCaml type checker do more work for us, extracting out a couple helpers, and rephrasing functions with pipes.

### CODE REVIEW

**Comment by expede** [comment]
This is me just playing around with the existing code. Nothing too radical here, mostly letting the OCaml type checker do more work for us.

File: src/Fission.re [hunk 1]
```diff
@@ -1,87 +1,111 @@
...
-Js.Promise.(
-  remove("QmQbPPkak9itW3v8WSohtonCBiJcrnAUhrSW1TGPnmWe3f", user)
-  |> then_(result => resolve(Js.Console.log(result)))
-  |> catch(error => resolve(Js.Console.log(error)))
-);
\ No newline at end of file
+let await = promise =>
+  promise
+  |> Js.Promise.then_(response => Js.Promise.resolve(response##data))
+  |> Js.Promise.catch(Js.Promise.resolve);
```
Comment (expede) [inline src/Fission.re]: This was repeated in a few places, so I just punned on JS's `await` name

File: src/Fission.re [hunk 2]
```diff
@@ -1,87 +1,111 @@
...
+
+let url = (domain, cid) => domain ++ "/ipfs/" ++ cid;
+let octetHeader = Axios.Headers.fromObj({"content-type": "application/octet-stream"});
+let octetConfig = auth => Axios.makeConfig(~auth=convAuth(auth), ~headers=octetHeader, ());
+let blankConfig = auth => Axios.makeConfig(~auth=convAuth(auth), ());
```
Comment (expede) [inline src/Fission.re]: Stripped out the type annotations. The type system goes out of its way to make this very stable, and to give you good warnings, and editor plugins will be able to tell you the type under the cursor (though sadly not in Spacemacs yet 🤷‍♀️)

File: src/Fission.re [hunk 3]
```diff
@@ -1,87 +1,111 @@
...
+
+// Main Show
+
+let content = cid =>
+  baseURL
+  -> url(cid)
+  -> Axios.get
+  -> await
```
Comment (expede) [inline src/Fission.re]: `->` is like dot-chaining, but threads values into the next function as the first argument. It makes it read as a series of steps, and has zero runtime overhead (i.e. you may as well use this if you have a chain of functions).

```reason
10
-> minusSix
-> add(5)
-> timesTen

// Exactly the same as
timesTen(add(minusSix(10), 5))
```

File: src/Fission.re [hunk 4]
```diff
@@ -1,87 +1,111 @@
...
+    url:    url(base)
+  };
+};
+
+let fission = base => {
+  base,
+  content,
+  login: fissionUser(base),
+  url:   url(base)
+};
```
Comment (expede) [inline src/Fission.re]: These two read really well, IMO. Very much like using bare JS objects rather than classes.

**Comment by dholms** [comment]
Closing because similar functionality added in: https://github.com/fission-suite/reasonml-client/pull/3

**Pull request closed without merging**